src/pybind/mgr/balancer/module.py

```diff
@@ -374,6 +374,14 @@ def handle_command(self, inbuf, command):
             }
             return (0, json.dumps(s, indent=4), '')
         elif command['prefix'] == 'balancer mode':
+            if command['mode'] == 'upmap':
+                min_compat_client = self.get_osdmap().dump().get('require_min_compat_client', '')
+                if min_compat_client < 'luminous': # works well because version is alphabetized..
+                    warn = 'min_compat_client "%s" ' \
+                           '< "luminous", which is required for pg-upmap. ' \
+                           'Try "ceph osd set-require-min-compat-client luminous" ' \
+                           'before enabling this mode' % min_compat_client
```

Comment:
not totally sure about suggesting they run the command here since there are bigger side-effects of changing the compat client setting.  @jdurgin what do you think?